cognite/neat/rules/exporter/rules2excel.py

```diff
@@ -5,62 +5,108 @@
 from openpyxl.cell import Cell
 from openpyxl.styles import Alignment, Border, Font, NamedStyle, PatternFill, Side
 
-from cognite.neat.rules.models import TransformationRules
+from cognite.neat.rules.models.raw_rules import RawRules
+from cognite.neat.rules.models.rules import Rules
 
+from ._base import BaseExporter
 
-class RulesToExcel:
+
+class ExcelExporter(BaseExporter):
     """Class for exporting transformation rules object to excel file."""
 
-    def __init__(self, rules: TransformationRules):
-        self.rules = rules
-        self.workbook = Workbook()
+    def __init__(self, rules: Rules | RawRules, filepath: Path, report_path: Path | None = None):
+        self.report = None
+        if rules.__class__.__name__ == RawRules.__name__:
+            self.rules = cast(RawRules, rules).to_rules(skip_validation=True)
+            self.report = cast(RawRules, rules).validate_rules()
+        else:
+            self.rules = cast(Rules, rules)
```

Comment:
Seems like something that should be handled in the base class

Reply:
Good catch !